graylog2-server/src/main/java/org/graylog2/indexer/indexset/IndexSetConfig.java

```diff
@@ -168,18 +76,19 @@ public static IndexSetConfig create(@Id @ObjectId @JsonProperty("_id") @Nullable
                                         @JsonProperty("index_wildcard") @Nullable String indexWildcard,
                                         @JsonProperty("shards") @Min(1) int shards,
                                         @JsonProperty("replicas") @Min(0) int replicas,
-                                        @JsonProperty("rotation_strategy_class") @Nullable String rotationStrategyClass,
-                                        @JsonProperty("rotation_strategy") @NotNull RotationStrategyConfig rotationStrategy,
-                                        @JsonProperty("retention_strategy_class") @Nullable String retentionStrategyClass,
-                                        @JsonProperty("retention_strategy") @NotNull RetentionStrategyConfig retentionStrategy,
+                                        @JsonProperty(FIELD_ROTATION_STRATEGY_CLASS) @Nullable String rotationStrategyClass,
+                                        @JsonProperty(FIELD_ROTATION_STRATEGY) @Nullable RotationStrategyConfig rotationStrategy,
+                                        @JsonProperty(FIELD_RETENTION_STRATEGY_CLASS) @Nullable String retentionStrategyClass,
+                                        @JsonProperty(FIELD_RETENTION_STRATEGY) @Nullable RetentionStrategyConfig retentionStrategy,
                                         @JsonProperty(FIELD_CREATION_DATE) @NotNull ZonedDateTime creationDate,
                                         @JsonProperty("index_analyzer") @Nullable String indexAnalyzer,
                                         @JsonProperty("index_template_name") @Nullable String indexTemplateName,
                                         @JsonProperty(FIELD_INDEX_TEMPLATE_TYPE) @Nullable String indexTemplateType,
                                         @JsonProperty("index_optimization_max_num_segments") @Nullable Integer maxNumSegments,
                                         @JsonProperty("index_optimization_disabled") @Nullable Boolean indexOptimizationDisabled,
                                         @JsonProperty("field_type_refresh_interval") @Nullable Duration fieldTypeRefreshInterval,
-                                        @JsonProperty("custom_field_mappings") @Nullable CustomFieldMappings customFieldMappings
+                                        @JsonProperty("custom_field_mappings") @Nullable CustomFieldMappings customFieldMappings,
+                                        @JsonProperty(FIELD_DATA_TIERS) @Nullable DataTieringConfig dataTiers
```

Comment:
This wont be able to deserialize [EnterpriseDataTieringConfig](https://github.com/Graylog2/graylog-plugin-enterprise/pull/6182/files#diff-502b47036657f6af19a5f7e82c61067814a5a3f8d2b145afdd513c60b411ab5e).
 
there is a get call to see which retention configuration is available. Maybe we need something like this? And have that part pluggable